.github/workflows/test-go-task.yml

```diff
@@ -197,6 +197,7 @@ jobs:
       - name: Send unit tests coverage to Codecov
         uses: codecov/codecov-action@v3
         with:
+          token: ${{ vars.CODECOV_TOKEN }}
```

Comment:
```suggestion

      # A token is used to avoid intermittent spurious job failures caused by rate limiting.
      - name: Set up Codecov upload token
        run: |
          if [[ "${{ github.repository }}" == "arduino/arduino-ide" ]]; then
            # In order to avoid uploads of data from forks, only use the token for runs in the arduino/arduino-ide repo.
            # Token is intentionally exposed.
            # See: https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954
            CODECOV_TOKEN="18b9885b-cbf2-4166-bfbb-39617323daf0"
          else
            # codecov/codecov-action does unauthenticated upload if empty string is passed via the `token` input.
            CODECOV_TOKEN=""
          fi

          echo "CODECOV_TOKEN=$CODECOV_TOKEN" >> "$GITHUB_ENV"

      - name: Send unit tests coverage to Codecov
        uses: codecov/codecov-action@v3
        with:
          token: ${{ env.CODECOV_TOKEN }}
```

There was an unexpected result of the last workflow run:

I investigated and found I lied in my previous comment:

> Note that GitHub wrote some incorrect information on that page:
>
>> Anyone with collaborator access to this repository can use these secrets and variables for actions. They are not passed to workflows that are triggered by a pull request from a fork.
>
> This is correct for secrets, but incorrect for variables. Variables are passed to workflows triggered by PRs from forks.

That was based on an experiment I ran, but I must have made a mistake in the experiment or my interpretation of the results because when I try it now I see that, indeed variables are not passed to runs triggered by PRs from forks. It doesn't make any sense that GitHub would impose that restriction because the variables are printed in the logs so anyone can see them so there is no reason to treat them the same as secrets. But that's how it works. I apologize for leading this PR down a dead end road.

---

So the only solution is to go back to hardcoding the token in the workflow, but configure the workflow to only use that token for runs in this repository.

Although I don't like that the new approach adds complexity to the workflow, the advantage is that the whole system is accessible to all contributors rather than relying on a repository variable that can only be set by an administrator.

Reply:
> `if [[ "${{ github.repository }}" == "arduino/arduino-ide" ]]; then`

I guess you intended `arduino/arduino-cli` here.

Reply:
> I guess you intended arduino/arduino-cli here.

Oops, yeah copy/paste error. 🤦